### DISCUSSION TITLE
Notification-permission check before every action required?

### DISCUSSION THREAD

**Discussion by philitell** [781flyingdutchman/background_downloader#393] [Q&A] [answered]
The Readme-File contains following statement:

> The downloader will check permission status before each action, e.g. will not show notifications unless permissions for notifications have been granted

 - Based on that sentence it would be fine to check notification-permission e.g. on startup and if the permission is granted we can configure the notification (Or can the notification be configured anyway - regardless of the permission?).
 - Or should the permission be checked "manually" before every action (like shown in the example).

Whats the recommended way?

**Comment by 781flyingdutchman** [comment, marked as answer]
The reason to check every time is that the user may have revoked or given the permission while using the app, using the settings menu. If you don't check, or only check on start-up, you won't know. That said, nothing will break if you don't check (because the OS will simply obey the actual permissions set) so if you prefer to do it only occasionally that's totally fine. For example, you could check before a large number of downloads start, or when your "downloads screen" is first created, or as you suggest only on start-up.

**Reply by philitell** [reply]
That makes sense - thank you very much!
